### PR TITLE
Feature/add error history module

### PR DESCRIPTION
**Pull request by andistorm**
requires:
* https://github.com/EVerest/everest-utils/pull/79
* https://github.com/EVerest/everest-core/pull/446
* https://github.com/EVerest/everest-framework/pull/133

does not require any longer:
* https://github.com/EVerest/everest-utils/pull/110
* https://github.com/EVerest/everest-ci/pull/10

### CODE REVIEW

**Comment by andistorm** [comment]
Check required ec-cli before merging

**Comment by andistorm** [comment]
Do not merge before setting correct dependencies in dependencies.yaml

**Comment by corneliusclaussen** [comment]
Converting to draft as it is waiting for other PRs and cannot be merged now. Also needs cleanup of history after dependent PRs are merged.

File: dependencies.yaml
Comment (andistorm) [inline dependencies.yaml:4]:
```suggestion
  git_tag: v0.9.1
```

File: dependencies.yaml
Comment (andistorm) [inline dependencies.yaml:69]:
```suggestion
  git_tag: 1604c8b
```

File: dependencies.yaml
Comment (andistorm) [inline dependencies.yaml:4]:
```suggestion
  git_tag: v0.10.0
```

**Review by hikinggrass — APPROVED** [review]

File: modules/ErrorHistory/tests/CMakeLists.txt
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:2]:
```suggestion
set(TARGET_NAME ${PROJECT_NAME}_module_error_history_tests)
add_executable(${TARGET_NAME})

```

File: modules/ErrorHistory/tests/CMakeLists.txt
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:3]:
```suggestion
target_sources(${TARGET_NAME}
```

File: modules/ErrorHistory/tests/CMakeLists.txt
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:14]:
```suggestion
target_link_libraries(${TARGET_NAME}
```

File: modules/ErrorHistory/tests/CMakeLists.txt
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:25]:
```suggestion
    catch_discover_tests(${TARGET_NAME})
```

File: modules/ErrorHistory/CMakeLists.txt
Comment (andistorm) [inline modules/ErrorHistory/CMakeLists.txt:36]:
```suggestion
if(EVEREST_CORE_BUILD_TESTING)
```

File: modules/ErrorHistory/tests/CMakeLists.txt
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:29]:
```suggestion
add_test(${TARGET_NAME} ${TARGET_NAME})
```

**Pull request merged**
